drivers/place/booking_check_in_helper.cr

```diff
@@ -0,0 +1,279 @@
+RAd5uIjVwVDLXqaOpvF3pxWfl+cf6PJtAE5y+nbabeTmrw//fJMank3o7qCXkFZR
+F0OJ2tmENwV+LPM8Gy3So8YP2nkOz4bryaGrxQ4eMA+K9+RiACVaKv+tNx/NbyMS
+e9gg504u0cwa60XjM5KUKrmT3RXpY4YIfUPZ1J4dAoGAB6jalDOiSJ2j2G57acn3
+PGTowwN5g9IEXko3IsVWr0qIGZLExOaZxaBXsLutc5KhY9ZSCsFbCm3zWdhgZ7GA
+083i3dj3C970iHA3RToVJJbbj56ltFNd/OGiTwQpLcTsB3iVSFWVDbpsceXacG5F
+JWfd0O0RyaOk6a5IVbm+jMsCgYBglxAOfY4LSE8y6SCM+K3e5iNNZhymgHYPdwbE
+xPMrWgpfab/Evi2dBcgofM+oLU663bAOspMeoP/5qJPGxnNtC7ZbSMZNL6AxBVj+
+ZoW3uHsMXz8kNL8ixecTIxiO5xlwltPVrKExL46hsCKYFhfzcWGUx4DULTLMBCFU
++M/cFQKBgQC+Ite962yJOnE+bjtSReOrvR9+I+YNGqt7vyRa2nGFxL7ZNIqHss5T
+VjaMgjzVJqqYozNT/74pE/b9UjYyMzO/EhrjUmcwriMMan/vTbYoBMYWvGoy536r
+4n455vizig2c4/sxU5yu9AF9Dv+qNsGCx2e9uUOTDUlHM9NXwxU9rQ==
+-----END RSA PRIVATE KEY-----
+STRING
+  })
+
+  def on_load
+    on_update
+  end
+
+  # See: https://crystal-lang.org/api/latest/Time/Format.html
+  @date_time_format : String = "%c"
+  @time_format : String = "%l:%M%p"
+  @date_format : String = "%A, %-d %B"
+  @timezone : Time::Location = Time::Location::UTC
+
+  protected getter! prompt_after : Time::Span
+  protected getter! present_from : Time::Span
+
+  getter? meeting_pending : Bool = false
+  getter? people_present : Bool = false
+  getter current_meeting : PlaceCalendar::Event? = nil
+
+  # This last meeting id we prompted
+  @prompted : String = ""
+
+  # The URLS we want to send to the user
+  @check_in_url : String = ""
+  @no_show_url : String = ""
+  @domain : String = ""
+
+  @jwt_private_key : String = ""
+
+  def on_update
+    @jwt_private_key = setting(String, :jwt_private_key)
+
+    @prompt_after = (setting?(Int32, :prompt_after) || 10).minutes
+    @present_from = (setting?(Int32, :present_from) || 5).minutes
+
+    @check_in_url = setting(String, :check_in_url)
+    @no_show_url = setting(String, :no_show_url)
+    @domain = URI.parse(@check_in_url).host.not_nil!
+
+    subscriptions.clear
+    system.load_complete do
+      bookings.subscribe(:current_booking) do |_sub, pending|
+        if pending.presence && pending != "null"
+          update_current(PlaceCalendar::Event.from_json(pending))
+        else
+          update_current(nil)
+        end
+      end
+      bookings.subscribe(:current_pending) { |_sub, pending| update_pending(pending == "true") }
+      bookings.subscribe(:presence) { |_sub, presence| update_presence(presence == "true") }
+
+      monitor("#{config.control_system.not_nil!.id}/guest/bookings/prompted") do |_sub, response|
+        prompt_response(**NamedTuple(id: String, check_in: Bool).from_json(response))
+      end
+    end
+
+    timezone = setting?(String, :time_zone).presence || config.control_system.not_nil!.timezone.presence
+    @timezone = Time::Location.load(timezone) if timezone
+
+    @date_time_format = setting?(String, :date_time_format) || "%c"
+    @time_format = setting?(String, :time_format) || "%l:%M%p"
+    @date_format = setting?(String, :date_format) || "%A, %-d %B"
```

Comment:
Any way of DRY'ing up the defaults here with [what's up on L20-23](https://github.com/PlaceOS/drivers/pull/248/files#diff-7a1a70194fe7daff1c570961b9403059238ddb00aa77477eef8ef6ff9b2f5e3aR20-R23)?